commons/src/main/java/io/aiven/kafka/connect/common/source/input/JsonTransformer.java

```diff
@@ -80,7 +81,7 @@ private Stream<Object> readJsonRecordsAsStream(final IOSupplier<InputStream> inp
             } catch (IOException e) {
                 LOGGER.error("Error closing BufferedReader: {}", e.getMessage(), e);
             }
-        });
+        }).skip(skipRecords);
```

Comment:
There's probably something simple and nice we can do here with *NOT* parsing lines into JSON that are destined to be skipped... I'm not sure how Spliterator allows us to deal with this!

Reply:
Again with kcon-25, we do not parse lines at all.